### PR TITLE
Adding documentation about double quote implementation

### PR DESCRIPTION
**Pull request by MitchellGale-BitQuill**
### Description
Adds documentation to datatypes.rst regarding implementation and functionality for double quotes being used as escape characters in some situations. See below table for example of implementation. See [https://dev.mysql.com/doc/refman/8.0/en/string-literals.html](https://dev.mysql.com/doc/refman/8.0/en/string-literals.html) mysql documentation.

```    
+-----------+-----------+-------------+-------------+---------------+---------------+
| 'hello'   | "world"   | '"hello"'   | "'world'"   | '''hello'''   | """world"""   |
|-----------+-----------+-------------+-------------+---------------+---------------|
| hello     | world     | "hello"     | 'world'     | 'hello'       | "world"       |
+-----------+-----------+-------------+-------------+---------------+---------------+
```
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: docs/user/general/datatypes.rst [hunk 1]
```diff
@@ -353,16 +353,15 @@ A string can also represent and be converted to date and time types (except to i
 String Data Types
 =================
 
-A string is a sequence of characters enclosed in either single or double quotes. For example, both 'text' and "text" will be treated as string literal. To use quote characters in a string literal, you can include double quotes within single quoted string or single quotes within double quoted string::
+A string is a sequence of characters enclosed in either single or double quotes. For example, both 'text' and "text" will be treated as string literal. To use quote characters in a string literal, you can two quotes of the same type as the enclosing quotes::
```
Comment (dai-chen) [inline docs/user/general/datatypes.rst:356]: > you can [] two quotes ...

any verb missing in-between?

**Review by dai-chen — APPROVED** [review]
Thanks for the changes!

File: docs/user/general/datatypes.rst (hunk 1 above)
Comment (MitchellGale-BitQuill) [reply docs/user/general/datatypes.rst]: Good catch. Fixed.

**Review by dai-chen — APPROVED** [review]

**Review by acarbonetto — APPROVED** [review]

**Comment by penghuo** [comment]
CI failed?
Could you merge the commits in case there is only one file change?

**Review by anirudha — APPROVED** [review]

**Comment by joshuali925** [comment]
I'll merge in, CI failing should be irrelevant to this PR

**Pull request merged**